packages/core/src/combo-box/ComboBox.tsx

```diff
@@ -389,6 +390,7 @@ export const ComboBox = forwardRef(function ComboBox<Item>(
           withBaseName(),
           {
             [withBaseName("focused")]: focusedState,
+            "saltPillInput-focused": focusedState,
```

Comment:
There was already a class in pillInput.css `.saltPillInput-focused`. which was not being called earlier even though the class was defined. And the focus state of combobox was wrong as per design.

Otherwise will need to pass the `focusedState` prop down to the pillInput component and activate the  `.saltPillInput-focused` class for focus. 

Let me know your thoughts @origami-z

Reply:
Couldn't work out what's wrong with focus state right now.
But we don't do cross component classname passing around, will need to use js to handle it.

Reply:
removed implementation from this branch. will be taken care in #3612